src/pydagmc/dagnav.py

```diff
@@ -184,9 +202,30 @@ def create_volume(self, global_id: Optional[int] = None) -> Volume:
         """Create a new empty volume set"""
         return Volume.create(self, global_id)
 
-    def create_surface(self, global_id: Optional[int] = None) -> Surface:
-        """Create a new empty surface set"""
-        return Surface.create(self, global_id)
+    def create_surface(
+        self,
+        global_id: Optional[int] = None,
+        filename: Optional[PathLike] = None
+    ) -> Surface:
+        """Create a new surface set.
+
+        Parameters
+        ----------
+        global_id : int, optional
+            The global ID of the surface. If None, a new ID will be generated.
+        filename : path-like, optional
+            The file to read from. If None, the surface will be created empty.
+
+        Returns
+        -------
+        Surface
+            The new surface set.
+
+        """
+        surface = Surface.create(self, global_id)
+        if filename is not None:
+            self.mb.load_file(str(filename), surface.handle)
```

Comment:
I still think we may need to restrict this to STL files (by file extension).  I'm not sure if/how it performs if another legitimate MOAB file format is given?

Reply:
Just added in a check against the file extension to make sure that only .stl files are allowed :heavy_check_mark: